packages/dds/tree/src/shared-tree-core/editManager.ts

```diff
@@ -646,76 +646,104 @@ export class EditManager<
 		return Math.max(max, localPath.length);
 	}
 
-	public addSequencedChange(
-		newCommit: Commit<TChangeset>,
+	/* eslint-disable jsdoc/check-indentation */
+	/**
+	 * Add a bunch of sequenced changes. A bunch is group of sequenced commits that have the following properties:
+	 * - They are not interleaved with messages from other DDSes in the container.
+	 * - They are all part of the same batch, which entails:
+	 *   - They are contiguous in sequencing order.
+	 *   - They are all from the same client.
+	 *   - They are all based on the same reference sequence number.
+	 *   - They are not interleaved with messages from other clients.
+	 */
+	/* eslint-enable jsdoc/check-indentation */
+	public addSequencedChanges(
+		newCommits: readonly GraphCommit<TChangeset>[],
```

Comment:
nit: Reduce redundant data required by this method:
```suggestion
		newCommits: readonly Omit<GraphCommit<TChangeset>, "sessionId">[],
```

Alternatively, you could:
1. Remove the `sessionId` parameter and just use `newCommits[0].sessionId`. 
2. Type the `newCommits` parameter as `[GraphCommit<TChangeset>, ...GraphCommit<TChangeset>]` so that it's type safe within this function (that also lets you remove the `assert(newCommits.length > 0, ...)`).
3. Use `assert(hasSome(newCommits), ...)` before the call site to `addSequencedChanges` in SharedTreeCore.
4. If you want, add an `assert(newCommit.sessionId === newCommits[0].sessionId, "Expected all commits in a batch to have the same session ID")` in each of the loops in this method. I think we have [debug asserts](https://github.com/microsoft/FluidFramework/pull/23552) now, if you want.

I prefer the latter FWIW, since it's fewer parameters and fewer redundant ways to access the same data.

Reply:
`GraphCommit` doesn't have `sessionId`. I changed it from `Commit` to `GraphCommit`. Are you sguggesting changing it back to `Commit`?

Reply:
Oh, I see! I didn't realize that, I was indeed thinking that GraphCommit had a `sessionId`. In that case, it's fine as is.